<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from moving_robot.xacro             | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="lab3_robot">
  <!-- materials -->
  <material name="blue">
    <color rgba="0 0 0.9 1"/>
  </material>
  <material name="yellow">
    <color rgba="1 1 0 1"/>
  </material>
  <material name="green">
    <color rgba="0 1 0 1"/>
  </material>
  <!-- ROBOT BUILD STARTS HERE -->
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.1 0.1 0.1"/>
      </geometry>
      <origin xyz="0 0 0"/>
      <material name="green"/>
    </visual>
  </link>
  <link name="link_1">
    <visual>
      <geometry>
        <box size="0 0.1 0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0 0"/>
      <material name="blue"/>
    </visual>
  </link>
  <link name="link_2">
    <visual>
      <geometry>
        <box size="1 0.1 0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.5 0 0"/>
      <material name="blue"/>
    </visual>
  </link>
  <link name="link_3">
    <visual>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
      <origin rpy="0 1.57 0" xyz="0.1 0 0"/>
      <material name="yellow"/>
    </visual>
  </link>
  <joint name="base_to_link_1" type="revolute">
    <parent link="base_link"/>
    <child link="link_1"/>
    <origin rpy="0.0 -0.0 0.0" xyz="0.0 0.0 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.1415" upper="3.1415" velocity="0.5"/>
  </joint>
  <joint name="link1_to_link_2" type="revolute">
    <parent link="link_1"/>
    <child link="link_2"/>
    <origin rpy="-1.57 -0.0 0.0" xyz="0.0 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="1000.0" lower="-3.14" upper="0" velocity="0.5"/>
  </joint>
  <joint name="link2_to_link_3" type="revolute">
    <parent link="link_2"/>
    <child link="link_3"/>
    <origin rpy="0.0 -0.0 0.0" xyz="1.0 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5"/>
  </joint>
</robot>

